### ISSUE TITLE
CUDA Memory

### ISSUE THREAD

**Issue by LargeFishPKU** [voyazici/orderless-rnn-classification#7]
Hi, in the second training phase, when set the batch size to 32 with 4 GPU, it always incurs "out of cuda memory" error. the memory of each GPU is 10GB. when I set the batch size to 8, then it works. Therefore, may I ask what GPU and how many of them did you use to train  the encoder and decoder ?

**Comment by voyazici** [comment]
Hi,
I don't remember exactly how many GPUs I used but I remember that the batch size was 32. So I just relaunched the second stage  with the batch size of 32 and I see that I have to use 2 GPUs to fit the training:
+-------------------------------+----------------------+----------------------+
|   0  GeForce GTX 108...  Off  | 00000000:09:00.0 Off |                  N/A |
| 43%   80C    P2   104W / 250W |   9075MiB / 11178MiB |     86%      Default |
+-------------------------------+----------------------+----------------------+
|   1  GeForce GTX 108...  Off  | 00000000:0A:00.0 Off |                  N/A |
| 32%   61C    P2    87W / 250W |   8131MiB / 11178MiB |     91%      Default |
+-------------------------------+----------------------+----------------------+

**Comment by LargeFishPKU** [comment]
Hi, 
Thanks for the reponse. Another question is the order of "preds_al"l and "labels_all" in the function "precision_recall_fscore_support",  because according to the [official website ](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.precision_recall_fscore_support.html), the first parameter should be the labels rather than preds, which is opposite to your codes.

**Comment by voyazici** [comment]
Yes, you are right. Actually I answered the same question [here](https://github.com/voyazici/orderless-rnn-classification/issues/5#issuecomment-683623312). It had to be the other way around but since the output is 
`recall, precision, ... = precision_recall_fscore_support(...`
not
`precision, recall, ...  = precision_recall_fscore_support(...`
the code outputs the correct precision and recall values.
